decode.go

```diff
@@ -103,6 +107,18 @@ func (dec *Decoder) Start(n int) error {
 		n = 1
 	}
 
+	if dec.sizeBuf == nil {
+		dec.sizeBuf = make([]byte, 4, 4)
```

Comment:
May you move it to `NewDecoder()`?
Also, third arguments for `make` is not required.